libcudacxx/include/cuda/__stream/stream_ref.h

```diff
@@ -60,6 +60,7 @@ class stream_ref
   //!
   //! For behavior of the default stream,
   //! @see //! https://docs.nvidia.com/cuda/cuda-runtime-api/stream-sync-behavior.html
+  [[deprecated("Use the constructor from cudaStream_t instead")]]
```

Comment:
Suggestion: I expect many users running into this, so let's maybe be even more helpful:
```suggestion
  [[deprecated("Using the default/null stream is generally discouraged. If you need to use it, please construct a  stream_ref from cudaStream_t{nullptr}")]]
```
Or any similar wording.